tiledb/sm/query/query_condition.cc

```diff
@@ -1473,45 +1506,170 @@ Status QueryCondition::apply_clause_sparse(
       return Status_QueryConditionError(
           "Cannot perform query comparison; Unsupported query "
           "conditional type on " +
-          clause.field_name_);
+          node->get_field_name());
   }
 
   return Status::Ok();
 }
 
+template <typename BitmapType>
+void QueryCondition::apply_tree_sparse(
+    const tdb_unique_ptr<ASTNode>& node,
+    const ArraySchema& array_schema,
+    ResultTile& result_tile,
+    std::vector<BitmapType>& result_bitmap) const {
+  // Case on the type of the tree node.
+  if (!node->is_expr()) {
+    // In the simple value node case, run the value evaluator function and
+    // return the result back.
+    Status s = apply_ast_node_sparse<BitmapType>(
+        node, array_schema, result_tile, result_bitmap);
+    if (!s.ok()) {
+      throw std::runtime_error(
+          "apply_tree_sparse: apply_ast_node_sparse failed.");
+    }
+  } else {
+    // For each child, declare a new result buffer, combine it all into the
+    // other one based on the combination op.
+    const QueryConditionCombinationOp& combination_op =
+        node->get_combination_op();
+    if (combination_op == QueryConditionCombinationOp::AND) {
+      std::fill(result_bitmap.begin(), result_bitmap.end(), 1);
+    } else if (combination_op == QueryConditionCombinationOp::OR) {
+      std::fill(result_bitmap.begin(), result_bitmap.end(), 0);
+    }
+    for (const auto& child : node->get_children()) {
+      std::vector<BitmapType> child_result_bitmap(result_bitmap.size(), true);
+      apply_tree_sparse(child, array_schema, result_tile, child_result_bitmap);
+      // Note that the multiplication operator (which functions
+      // effectively as a bitwise AND) is used to combine the accumulator result
+      // buffer and child result buffer in the AND combination op case, and
+      // bitwise OR in the OR combination op case.
+      switch (combination_op) {
+        case QueryConditionCombinationOp::AND: {
+          for (uint64_t c = 0; c < result_bitmap.size(); ++c) {
+            result_bitmap[c] *= child_result_bitmap[c];
```

Comment:
Good question.  The previous code had `*=`.  I think one reason might be that the bitmap might actually be a bytemap (using `uint8_t` or `std::byte` rather than bool) since bytes can actually be much faster than bits.  Not sure but maybe `*=` is faster in the case of bytes?  That raises the question about using `|=` in the other case, though `+=` wouldn't be equivalent, so maybe there isn't an alternative to `|=`.  This is a lot of speculation though.

Reply:
Yea, I'd agree with the good question too. It wasn't explained and was part of the original PR, so I kept it as is. I was thinking of running a quick performance test to see which one actually was faster though.

Reply:
Let's add a comment then in the code that this should be tested for performance

Reply:
Andrew and I have been working on testing for performance and have reached a point where there aren't performance deficits from using the code, so I think not adding the comment is fine at this point.

Reply:
Well, we haven’t actually done the & vs * comparisons.  I don’t think there will be a difference - though on the M1 which is compute bound rather than memory bound we might see something (if there is something to see).  So I’d say we should leave in a comment about verifying & vs *.  I’ll try to find some time to actually do the experiment.